design_minstack.py

```diff
@@ -0,0 +1,41 @@
+# Time Complexity :push:- O(1) pop:-O(1) top- O(1) getmin:- O(1)
+# Space Complexity : O(n)
+# Did this code successfully run on Leetcode : Yes
+# Any problem you faced while coding this : At first I thought of using a heap to keep track of the minimum element but that would give me O(logn) time
```

Comment:
Apart from that, the interviewer expects you to use only stacks here

Reply:
@kannan-mrugapphan In python, I don't think we have a stack data structure. As far as I understood a stack doesn't use contiguous memory locations which an array which many people call stack in python does. The closest python comes to having a stack data structure is the deque in the collections framework which can be called a stack if don't use the popleft() method. Otherwise, I have to implement my own stack data structure. Please let me know your thoughts about this.